multicast/discovery.go

```diff
@@ -134,6 +137,25 @@ func (d *Discovery) Respond(ctx context.Context, info ServerInfo) error {
 	return nil
 }
 
+// StopResponder stops the responder server and cancels it's inner context.
+func (d *Discovery) StopResponder() error {
+	// Check if this instance of discovery has an active responder server connection.
+	if d.responderConn != nil {
```

Comment:
Calling `StopResponder` without a prior call to `Respond` may result in a nil `d.responderCancel`, which would cause a panic if `d.responderCancel()` is called. To avoid that, can we add a nil check for `d.responderCancel` dow below before calling it:
```go
if d.responderCancel != nil {
    d.responderCancel()
}
```

Reply:
Fixed